### PR TITLE
Migrate from IcCDK to IcKit

### PR DESCRIPTION
**Pull request by qti3e**
This pull request removes the usage of ic_cdk in the codebase and replaces it with ic_kit to allow more unit testing in future.

### CODE REVIEW

**Comment by kanej** [comment]
I am getting an error on deploying piggy-bank:

image

File: xtc/src/history.rs [hunk 1]
```diff
@@ -50,7 +49,7 @@ impl HistoryBuffer {
                 return 0;
             }
 
-            trap("Transaction is expected to have a non-zero amount.")
```
Comment (ferencdg) [inline xtc/src/history.rs:53]: not sure about this change: api::trap is defined in the cdk and currently calls into ic0::trap which seems to be a C function imported into the webassembly module.

do we want to assume that panic! is doing the same thing as trap calling into ic0::trap? Also what if at some point the cdk changes api::trap's implementation to do something before calling into ic0::trap

sorry if I am being too cautious here

Comment (qti3e) [reply xtc/src/history.rs:53]: panic also traps the canister, they have the same effect.

**Review by kanej — APPROVED** [review]
I was able to complete the manual testing suite and the 300 transaction test of archive flushing (+ subsequent manual calls).

I had to update my rust version with `rustup update` first.

**Pull request merged**